go/kbfs/libkbfs/favorites.go

```diff
@@ -342,7 +350,40 @@ func (f *Favorites) handleReq(req *favReq) (err error) {
 	// quickly when offline.
 	needFetch := (req.refresh || f.favCache == nil) && !req.clear
 	wantFetch := f.config.Clock().Now().After(f.cacheExpireTime) && !req.clear
-	changed := false
+
+	for _, fav := range req.toAdd {
+		// FOR DISCUSSION: we don't check this in toDel because it could be
+		// out-of-date. So perhaps we should get rid of this too? Relevant
+		// question, if a favorite is deleted from another device, do we get
+		// notification from core for cache invalidation?
+		_, present := f.favCache[fav.Folder]
+		if !fav.Created && present {
+			continue
+		}
+		//
+		err := kbpki.FavoriteAdd(req.ctx, fav.ToKBFolder())
+		if err != nil {
+			f.log.CDebugf(req.ctx,
+				"Failure adding favorite %v: %v", fav, err)
+			return err
+		}
+		needFetch = true
+		changed = true
+	}
+
+	for _, fav := range req.toDel {
+		// Since our cache isn't necessarily up-to-date, always delete
+		// the favorite.
+		folder := fav.ToKBFolder(false)
+		err := kbpki.FavoriteDelete(req.ctx, folder)
+		if err != nil {
+			return err
+		}
+		f.config.UserHistory().ClearTLF(tlf.CanonicalName(fav.Name), fav.Type)
+		needFetch = true
```

Comment:
Why do we need to fetch in this case?  Don't we only need to fetch in order to get the mtime for favorited TLFs?

Reply:
I was trying to avoid editing the cache directly and always derive it from an RPC to core. I feel like it's easier to reason about. This just goes to service and is not expensive anyway right?

Reply:
I see.  I guess it's not expensive for regular users, but might be for something like keybasepub if it ever deleted favorites.  Plus it just adds another round trip time for no reason.  I think I'd prefer to just delete it from the cache here and not refresh if there was no other change.  I can't see how that would cause any problems.

Reply:
OK will do